src/rust/src/declarative_asn1/encode.rs

```diff
@@ -87,6 +90,27 @@ impl asn1::Asn1Writable for AnnotatedTypeObject<'_> {
                         .ok_or(asn1::WriteError::AllocationError)?;
                 write_value(writer, &printable_string)
             }
+            Type::UtcTime() => {
+                let val: &pyo3::Bound<'_, UtcTime> = value
+                    .downcast()
+                    .map_err(|_| asn1::WriteError::AllocationError)?;
+                let datetime_info = get_datetime_info(py, &val.get().inner)
+                    .map_err(|_| asn1::WriteError::AllocationError)?;
```

Comment:
Similarly, we have `py_to_datetime`, but I think we want an impl using the pyo3 datetime types.

Reply:
the tzinfo field is available though: https://docs.rs/pyo3/latest/pyo3/types/trait.PyTzInfoAccess.html